File: output/GPT3.5-Turbo/xCodeEval_4000/Java/C/f6fd3be8d47a284e826f131999a93862.c
#include <stdio.h>
#include <stdlib.h>

#define MOD 1000000007
#define MAXN 100

int gcd(int a, int b) {
    if (a < b) {
        int temp = b;
        b = a;
        a = temp;
    }

    while (b > 0) {
        int r = a % b;
        a = b;
        b = r;
    }
    return a;
}

void solve() {
    int n;
    scanf("%d", &n);

    long long ans = 0;
    long long check[MAXN + 1] = {0};

    int primeFactors[MAXN + 1][MAXN + 1] = {0};

    for (int i = 2; i <= n; i++) {
        if (primeFactors[i][0] == 0) {
            primeFactors[i][0] = i;

            if (n / i >= i) {
                for (int j = i * i; j <= n; j += i) {
                    if (primeFactors[j][0] == 0) {
                        primeFactors[j][0] = i;
                    }
                }
            }
        } else {
            int prime = primeFactors[i][0];
            int rest = i / prime;
            while (rest % prime == 0) {
                rest /= prime;
            }
            if (rest > 1) {
                for (int k = 0; k < MAXN + 1; k++) {
                    if (primeFactors[rest][k] == 0) {
                        primeFactors[i][k] = prime;
                        break;
                    }
                }
            }
        }
    }

    for (long long g = 1; g <= n; g++) {
        if (g * 2 + 1 > n) {
            break;
        }
        long long cnt = 0;
        long long sum = 0;

        for (int ap = 2; ap <= n; ap++) {
            if (g * (ap + 1) + 1 > n) {
                break;
            }

            long long maxBp = (ap - 1) < ((n - 1) / g - ap) ? (ap - 1) : ((n - 1) / g - ap);
            long long currCnt = maxBp;
            sum += (maxBp * (maxBp + 1) / 2) % MOD;

            int k = 0;
            while (primeFactors[ap][k] != 0) {
                long long curr = 1;
                int sign = 1;
                for (int pos = 0; pos < MAXN + 1; pos++) {
                    if (primeFactors[ap][pos] == 0) {
                        break;
                    }
                    if ((k & (1 << pos)) > 0) {
                        curr *= primeFactors[ap][pos];
                        sign *= -1;
                    }
                }

                long long s = (ap - 1) / curr < (n - 1 - g * ap) / curr / g ? (ap - 1) / curr : (n - 1 - g * ap) / curr / g;

                currCnt += s * sign;
                sum += curr * s * (s + 1) / 2 * sign;

                k++;
            }

            cnt += currCnt;
            sum += currCnt * ap;
            sum %= MOD;
        }

        cnt *= 2;
        sum *= g * 2;

        cnt++;
        sum += g * 2;

        cnt %= MOD;
        sum %= MOD;
        if (sum < 0) {
            sum += MOD;
        }

        check[g] -= ans;
        long long k = gcd(g, n);
        ans += g * n / k * cnt % MOD;
        ans -= g / k * sum % MOD;
        check[g] += ans;
    }

    ans %= MOD;
    if (ans < 0) {
        ans += MOD;
    }

    printf("%lld\n", ans);
}

int main() {
    int t = 1;

    for (int i = 1; i <= t; ++i) {
        solve();
    }

    return 0;
}

File: output/GPT4o-Mini/xCodeEval_4000/Java/Python/f6fd3be8d47a284e826f131999a93862.py
import sys
import random
from collections import defaultdict
from math import gcd

IMPOSSIBLE = "IMPOSSIBLE"
POSSIBLE = "POSSIBLE"
YES = "YES"
NO = "NO"
MOD = 1_000_000_007

class MyScanner:
    def __init__(self, input_stream):
        self.input_stream = input_stream

    def next_int(self):
        return int(self.input_stream.readline().strip())

    def next_int_array(self, length):
        return list(map(int, self.input_stream.readline().strip().split()))[:length]

class MyPrintWriter:
    def __init__(self, output_stream):
        self.output_stream = output_stream

    def println(self, value):
        self.output_stream.write(str(value) + "\n")

    def print_ans(self, arr):
        if arr:
            self.output_stream.write(" ".join(map(str, arr)) + "\n")

class Round818E:
    def __init__(self):
        self.in_stream = sys.stdin
        self.out_stream = sys.stdout
        self.in_scanner = MyScanner(self.in_stream)
        self.out_writer = MyPrintWriter(self.out_stream)
        self.n = 0
        self.ans = 0
        self.check = []

    def get_input(self):
        self.n = self.in_scanner.next_int()

    def print_output(self):
        self.out_writer.println(self.ans)

    def solve(self):
        prime_factors = [[] for _ in range(self.n + 1)]
        for i in range(2, self.n + 1):
            if not prime_factors[i]:
                prime_factors[i].append(i)
                for j in range(i * i, self.n + 1, i):
                    if not prime_factors[j]:
                        prime_factors[j].append(i)
            else:
                prime = prime_factors[i][0]
                rest = i // prime
                while rest % prime == 0:
                    rest //= prime
                if rest > 1:
                    prime_factors[i].extend(prime_factors[rest])

        self.ans = 0
        self.check = [0] * (self.n + 1)
        for g in range(1, self.n + 1):
            if g * 2 + 1 > self.n:
                break
            cnt = 0
            sum_ = 0

            for ap in range(2, self.n + 1):
                if g * (ap + 1) + 1 > self.n:
                    break

                max_bp = min(ap - 1, (self.n - 1) // g - ap)
                curr_cnt = max_bp
                sum_ += max_bp * (max_bp + 1) // 2 % MOD

                k = len(prime_factors[ap])
                for mask in range(1, 1 << k):
                    curr = 1
                    sign = 1
                    for pos in range(k):
                        if (mask & (1 << pos)) > 0:
                            curr *= prime_factors[ap][pos]
                            sign *= -1

                    s = min((ap - 1) // curr, (self.n - 1 - g * ap) // curr // g)

                    curr_cnt += s * sign
                    sum_ += curr * s * (s + 1) // 2 * sign

                cnt += curr_cnt
                sum_ += curr_cnt * ap
                sum_ %= MOD

            cnt *= 2
            sum_ *= g * 2

            cnt += 1
            sum_ += g * 2

            cnt %= MOD
            sum_ %= MOD
            if sum_ < 0:
                sum_ += MOD

            self.check[g] -= self.ans
            k = gcd(g, self.n)
            self.ans += g * self.n // k * cnt % MOD
            self.ans -= g // k * sum_ % MOD
            self.check[g] += self.ans

        self.ans %= MOD
        if self.ans < 0:
            self.ans += MOD

    def run(self):
        t = 1
        for _ in range(t):
            self.get_input()
            self.solve()
            self.print_output()

if __name__ == "__main__":
    sol = Round818E()
    sol.run()
